source/material_model/visco_plastic.cc

```diff
@@ -532,7 +611,12 @@ namespace aspect
       // Loop through all requested points
       for (unsigned int i=0; i < in.temperature.size(); ++i)
         {
-          // First compute the equation of state variables and thermodynamic properties
+
+          const std::vector<double> phase_fractions = compute_phase_fractions(in.position[i],
+                                                                              in.temperature[i],
+                                                                              in.pressure[i]);
```

Comment:
Just to make sure: This limits you to having the same phase transitions for all compositional fields. Is that what you want?

Reply:
No that is not what we want this should be composition dependent.

Reply:
This links into the question regarding how to handle more than three phases. Currently, when the calculation of the phase function is indeed composition independent.

However, one can have the material properties stay for a given compositional field if that material should not experience a given phase transition. For example, the eclogite phase transition is not relevant to mantle compositions, so one could specify that the density (or any other properties) of 'mantle' does not change during this phase transition.

If we want the phase functions to be composition dependent, then we need to have an input table for the phase transition width/pressures as well and the phase function algorithm needs to be re-written.

Does this make sense and do you think we should handle this in the current PR? I still need to finish moving `compute_phase_function`into `MaterialUtilities`, but the current structure would not work for composition-dependent phase functions.